Jenkinsfile

```diff
@@ -26,35 +26,36 @@ node('docker&&linux') {
 
     timestamps {
         stage('Generate Plugin Data') {
-            docker.image('maven').inside {
-              sh 'mvn -PgeneratePluginData'
-            }
+          infra.runWithMaven(
+            'mvn --no-transfer-progress -PgeneratePluginData',
+            /*jdk*/ '17',
+            /*extraEnv*/ null,
+            /*addToolEnv*/ false
+          )
         }
 
         /*
          * Running everything within an nginx container to provide the
          * DATA_FILE_URL necessary for the build and execution of the docker
          * container
          */
-        docker.image('nginx:alpine').withRun('-v $PWD/target:/usr/share/nginx/html') { c ->
+        docker.image('nginx:alpine').withRun('-p 80:80 -v $PWD/target:/usr/share/nginx/html') { c ->
 
             /*
              * Building our war file inside a Maven container which links to
              * the nginx container for accessing the DATA_FILE_URL
              */
             stage('Build') {
-                docker.image('maven:3-adoptopenjdk-11').inside("--link ${c.id}:nginx") {
-                    withEnv([
-                        'DATA_FILE_URL=http://nginx/plugins.json.gzip',
-                    ]) {
-                        infra.runWithMaven(
-                            'mvn -Dmaven.test.failure.ignore verify',
-                            /*jdk*/ "8",
-                            /*extraEnv*/ null,
-                            /*addToolEnv*/ false
-                          )
-                    }
-                }
+              withEnv([
+                'DATA_FILE_URL=http://localhost/plugins.json.gzip',
+              ]) {
+                infra.runWithMaven(
+                    'mvn --no-transfer-progress -Dmaven.test.failure.ignore verify',
```

Comment:
```suggestion
                    'mvn --no-transfer-progress -Dmaven.test.failure.ignore verify --version',
```

Reply:
https://ci.jenkins.io/job/Infra/job/plugin-site-api/job/PR-120/11/pipeline-console/?start-byte=0&selected-node=35#log-0 (build stage) reports the following:

```
+ mvn --no-transfer-progress -Dmaven.test.failure.ignore verify --version
  -XX:ConcGCThreads=1 -XX:G1ConcRefinementThreads=4 -XX:GCDrainStackTargetSize=64 -XX:InitialHeapSize=261829504 -XX:MarkStackSize=4194304 -XX:MaxHeapSize=4189272064 -XX:MinHeapSize=6815736 -XX:+PrintCommandLineFlags -XX:ReservedCodeCacheSize=251658240 -XX:+SegmentedCodeCache -XX:+UseCompressedClassPointers -XX:+UseCompressedOops -XX:+UseG1GC
  Apache Maven [3](https://ci.jenkins.io/job/Infra/job/plugin-site-api/job/PR-120/11/pipeline-console/?start-byte=0&selected-node=35#log-3).9.4 (dfbb324ad4a7c8fb0bf182e6d91b0ae20e3d2dd9)
  Maven home: /usr/share/apache-maven-3.9.[4](https://ci.jenkins.io/job/Infra/job/plugin-site-api/job/PR-120/11/pipeline-console/?start-byte=0&selected-node=35#log-4)
  Java version: 17.0.8.1, vendor: Eclipse Adoptium, runtime: /opt/jdk-17
```

It is an issue ith the `infra.runWithMaven()` function which does not seem to use the specified JDK8 :thinking:

Reply:
https://ci.jenkins.io/job/Infra/job/plugin-site-api/job/PR-120/11/pipeline-console/?start-byte=0&selected-node=35#log-0 (build stage) reports the following:

```
+ mvn --no-transfer-progress -Dmaven.test.failure.ignore verify --version
  -XX:ConcGCThreads=1 -XX:G1ConcRefinementThreads=4 -XX:GCDrainStackTargetSize=64 -XX:InitialHeapSize=261829504 -XX:MarkStackSize=4194304 -XX:MaxHeapSize=4189272064 -XX:MinHeapSize=6815736 -XX:+PrintCommandLineFlags -XX:ReservedCodeCacheSize=251658240 -XX:+SegmentedCodeCache -XX:+UseCompressedClassPointers -XX:+UseCompressedOops -XX:+UseG1GC
  Apache Maven [3](https://ci.jenkins.io/job/Infra/job/plugin-site-api/job/PR-120/11/pipeline-console/?start-byte=0&selected-node=35#log-3).9.4 (dfbb324ad4a7c8fb0bf182e6d91b0ae20e3d2dd9)
  Maven home: /usr/share/apache-maven-3.9.[4](https://ci.jenkins.io/job/Infra/job/plugin-site-api/job/PR-120/11/pipeline-console/?start-byte=0&selected-node=35#log-4)
  Java version: 17.0.8.1, vendor: Eclipse Adoptium, runtime: /opt/jdk-17
```

It is an issue ith the `infra.runWithMaven()` function which does not seem to use the specified JDK8 :thinking: